wp-content/civi-extensions/civirazorpay/CRM/Core/Civirazorpay/Payment/Razorpay.php

```diff
@@ -462,6 +466,21 @@ private function processSubscriptionCharged(array $event) {
 
       $pendingContribution = $this->getPendingContributionByRecurId($recurringContribution['id']);
 
+      $sourceField = CustomField::get(FALSE)
+        ->addSelect('id')
+        ->addWhere('custom_group_id:name', '=', 'Contribution_Details')
+        ->addWhere('name', '=', 'PAN_Card_Number')
+        ->execute()->single();
+
+      $sourceFieldId = 'custom_' . $sourceField['id'];
+
+      $campaigns = Campaign::get(FALSE)
+        ->addSelect('id')
+        ->addWhere('title', '=', $campaignTitle)
```

Comment:
Is this where clause good enough? Is campaign title unique?

Reply:
Yes, titles are uniques ther,e and on Razorpay Notes, we are only get the title of campaign and on based of that we are fetching the id of campaign